src/main/java/org/ladysnake/blabber/impl/common/illustrations/DialogueIllustrationCollection.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Blabber
+ * Copyright (C) 2022-2024 Ladysnake
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; If not, see <https://www.gnu.org/licenses>.
+ */
+package org.ladysnake.blabber.impl.common.illustrations;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.gui.DrawContext;
+import net.minecraft.entity.Entity;
+import net.minecraft.server.command.ServerCommandSource;
+import org.jetbrains.annotations.Nullable;
+import org.ladysnake.blabber.api.DialogueIllustration;
+import org.ladysnake.blabber.api.DialogueIllustrationType;
+import org.ladysnake.blabber.impl.common.BlabberRegistrar;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public record DialogueIllustrationCollection(List<DialogueIllustration> sub) implements DialogueIllustration {
+    private static final Codec<DialogueIllustrationCollection> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+            Codec.list(DialogueIllustrationType.CODEC).fieldOf("sub").forGetter(DialogueIllustrationCollection::sub)
+    ).apply(instance, DialogueIllustrationCollection::new));
+
+    public static final DialogueIllustrationType<DialogueIllustrationCollection> TYPE = new DialogueIllustrationType<>(
+            CODEC,
+            buf -> new DialogueIllustrationCollection(buf.readList(b -> {
+                DialogueIllustrationType<?> type = b.readRegistryValue(BlabberRegistrar.ILLUSTRATION_REGISTRY);
+                assert type != null;
+                return type.readFromPacket(b);
+            })),
+            (buf, item) ->
+                buf.writeCollection(item.sub, (b, i) -> {
+                    // Write the type, then the packet itself.
+                    b.writeRegistryValue(BlabberRegistrar.ILLUSTRATION_REGISTRY, i.getType());
+                    i.getType().writeToPacketUnsafe(b, i);
+                })
```

Comment:
Maybe the illustration list reading and writing should be a static method somewhere ? Since it's already used in two places

Reply:
Nah, it's only used in one place, it just got moved

Reply:
Ah alright, missed that